varats-core/varats/experiment/steps/ifthenelse.py

```diff
@@ -0,0 +1,72 @@
+import textwrap
+import typing as tp
+from pathlib import Path
+
+from benchbuild.utils.actions import StepResult, ProjectStep
+
+from benchbuild import Project
+from varats.experiment.experiment_util import OutputFolderStep
+
+
+def _run_with_output_if_necessary(
+    tmp_dir: Path, step: ProjectStep
+) -> StepResult:
+    if isinstance(step, OutputFolderStep):
+        return step.call_with_output_folder(tmp_dir)
+    else:
+        return step()
+
+
+class IfThenElse(OutputFolderStep):
+    """
+    Step that executes one of two sub steps based on a condition.
+
+    This step implements both interfaces for ProjectSteps and OutputFolderSteps.
+    """
+
+    def __init__(
+        self,
+        project: Project,
+        condition: ProjectStep,
+        if_step: tp.Optional[ProjectStep] = None,
+        else_step: tp.Optional[ProjectStep] = None
+    ) -> None:
+        super().__init__(project)
+        self.__condition = condition
+        self.__if_step = if_step
+        self.__else_step = else_step
+
+    def call_with_output_folder(self, tmp_dir: Path) -> StepResult:
+        result = _run_with_output_if_necessary(tmp_dir, self.__condition)
+
+        run_step = self.__if_step if result == StepResult.OK or result == StepResult.CAN_CONTINUE else self.__else_step
```

Comment:
This condition can be simplified by checking if the result is in a list of allowed results.
With that: do we want to make this list a parameter of this step? The default would be the same as now, but one could easily customize the condition that way.

```suggestion
        run_step = self.__if_step if result in [StepResult.OK, StepResult.CAN_CONTINUE] else self.__else_step
```